### ISSUE TITLE
Enhancement to allow for crumb trails with dropdown menus

### ISSUE THREAD

**Issue by nickludlam** [diericx/svelte-breadcrumbs#6]
Hey @diericx, this isn't a bug, more of a feature request. I'm implementing a breadcrumb nav with some of the crumbs mapping to dropdown menus.

The absolute easiest one-line fix for this is to add an `alternatives` array to Crumb, and make it self-referential:

```ts
export type Crumb = {
    title?: string;
    url?: string;
    alternatives?: Crumb[];
};
```

Does this seem reasonable enough to add in?

**Comment by diericx** [comment]
Hello! Could you describe this solution a bit further? What exactly is `alternatives` for and how does it end up getting used?

**Comment by nickludlam** [comment]
Sure, these are not my examples, just indicative of what I'm using this feature for:

https://flowbite.com/docs/components/breadcrumb/#breadcrumb-with-dropdown
https://nextui.org/docs/components/breadcrumbs#custom-items
https://codepen.io/kkhenriquez/pen/PQxvaa

In my specific case, the resources I'm mapping to are chapters of a book, so I'd like a convenient and compact way of changing the current nav depth to any chapter via a dropdown within the breadcrumbs, like this:

![image](https://github.com/diericx/svelte-breadcrumbs/assets/7568/06dda309-1aa3-45c8-be23-e329b807f320)

**Comment by diericx** [comment]
Are there SvelteKit routes for each chapter, or is there a route to a particular book and the chapter breadcrumb drop down is generated from the data for the book?

**Comment by diericx** [comment]
Edit: Sorry this doesn't work, it doesn't have the page data... will wait for the reply above before brainstorming just to be sure

**Comment by nickludlam** [comment]
> Are there SvelteKit routes for each chapter, or is there a route to a particular book and the chapter breadcrumb drop down is generated from the data for the book?

Yes, right now for the specific routes that have alternates, I'm constructing them on the server and feeding them down to the layout along with any other page data:

```ts
  const alternativeCrumbs = await getBookChapterAlternativeCrumbs(chapter.book.id);

  const crumbs = [
    { title: 'Books', url: '/books' },
    { title: chapter.book.title), url: `/books/${chapter.book.id}` },
    { title: chapter.title, url: `/books/${chapter.book.id}/chapters/${chapter.id}`, alternatives: alternativeCrumbs },
  ] as Crumb[];

  return { chapter, chapterHTML, crumbs };
```

And then the renderer becomes:

```svelte

<Breadcrumbs
  url={$page.url}
  routeId={$page.route.id}
  pageData={$page.data}
  crumbs={$page.data.crumbs}
  let:crumbs
>
...
```

**Comment by diericx** [comment]
Here is a PR I just opened up. It is a more generic solution that allows adding metadata to the `Crumb` array. https://github.com/diericx/svelte-breadcrumbs/pull/7

I think this is progress towards a good solution, but there is a pretty serious issue that I detailed out in the description. I'd love to know what you think and if you have any ideas for a better solution!

**Comment by diericx** [comment]
Actually I just found a better solution and updated that PR, check it out and lmk what you think

**Comment by nickludlam** [comment]
That's great, seems like a more flexible approach to let people add whatever data they need!

**Comment by diericx** [comment]
Great! I'll consider this closed with #7.

A new release has been pushed to npm as `1.2.4`
